include/swift/AST/DiagnosticsSema.def

```diff
@@ -2972,18 +2972,18 @@ NOTE(protocol_witness_missing_differentiable_attr,none,
 // @derivative
 ERROR(derivative_attr_expected_result_tuple,none,
       "'@derivative(of:)' attribute requires function to return a two-element "
-      "tuple of type '(value: T..., pullback: (U.TangentVector) -> T.TangentVector...)' "
-      "or '(value: T..., differential: (T.TangentVector...) -> U.TangentVector)'", ())
+      "tuple; first element must have label 'value:' and second element must "
+      "have label 'pullback:' or 'differential:'", ())
 ERROR(derivative_attr_invalid_result_tuple_value_label,none,
       "'@derivative(of:)' attribute requires function to return a two-element "
-      "tuple (first element must have label 'value:')", ())
+      "tuple; first element must have label 'value:'", ())
```

Comment:
Note: `derivative_attr_expected_result_tuple` (1) is somewhat duplicated by `derivative_attr_invalid_result_tuple_value_label` (2) and `derivative_attr_invalid_result_tuple_func_label` (3).

(1) is produced when the `@derivative` function type doesn't return a two-element tuple. (2) and (3) are produced when the returned two-element tuple's labels are incorrect.

Unless someone has suggestions, I'm content to leave (2) and (3) as is, since they're more specific than (1).